File: examples_android/app/src/main/java/com/confirmit/testsdkapp/layouts/defaultlayoutjava/cells/DefaultLayoutClickLabelCell.java
package com.confirmit.testsdkapp.layouts.defaultlayoutjava.cells;

import android.view.View;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.constraintlayout.widget.ConstraintLayout;
import androidx.constraintlayout.widget.ConstraintSet;
import androidx.core.content.ContextCompat;

import com.confirmit.testsdkapp.AppConfigs;
import com.confirmit.testsdkapp.R;
import com.confirmit.testsdkapp.layouts.defaultlayoutjava.items.DefaultLayoutClickLabel;
import com.confirmit.testsdkapp.layouts.defaultlayoutjava.items.DefaultLayoutClickLabelType;
import com.confirmit.testsdkapp.layouts.defaultlayoutjava.items.DefaultLayoutItem;

public class DefaultLayoutClickLabelCell extends DefaultLayoutCell implements View.OnClickListener {
    private DefaultLayoutCellChangeListener listener;
    private DefaultLayoutClickLabel item;
    private Boolean itemSelected;
    private ConstraintLayout btnContainer;
    private TextView lblText;
    private ImageView imgIcon;

    public DefaultLayoutClickLabelCell(View view) {
        super(view);
        itemSelected = false;
        btnContainer = view.findViewById(R.id.containerButton);
        btnContainer.setOnClickListener(this);
        lblText = view.findViewById(R.id.txtLabel);
        imgIcon = view.findViewById(R.id.imgIcon);
    }

    @Override
    public void setup(DefaultLayoutItem item, DefaultLayoutCellChangeListener listener) {
        this.item = (DefaultLayoutClickLabel) item;
        this.listener = listener;

        if (this.item.getAnswer().getText().getHasStyle() && AppConfigs.INSTANCE.getHtmlText()) {
            lblText.setText(this.item.getAnswer().getText().getSpanned());
        } else {
            lblText.setText(this.item.getAnswer().getText().get());
        }

        float density = lblText.getResources().getDisplayMetrics().density;
        ConstraintSet set = new ConstraintSet();
        set.clone(btnContainer);
        set.setMargin(imgIcon.getId(), ConstraintSet.START, (int) (density * 24 + this.item.getIndent() * (density * 24)));
        set.applyTo(btnContainer);

        itemSelected = listener.getSelected(this.item.getAnswer());

        setIcon();
    }

    @Override
    public void onClick(View view) {
        if (itemSelected && item.getClickLabelType() == DefaultLayoutClickLabelType.RADIO) {
            return;
        }

        itemSelected = !itemSelected;
        setIcon();
        switch (item.getClickLabelType()) {
            case CHECKBOX:
                listener.onCheckClicked(item.getAnswer(), itemSelected);
                break;
            case RADIO:
                listener.onRadioClicked(item.getAnswer());
                break;
        }
    }

    private void setIcon() {
        int imageId;
        switch (item.getClickLabelType()) {
            case CHECKBOX:
                imageId = itemSelected ? R.drawable.ic_outline_check_black : R.drawable.ic_outline_box_black;
                break;
            default:
                imageId = itemSelected ? R.drawable.ic_radio_black : R.drawable.ic_radio_unchecked_black;
                break;
        }

        imgIcon.setImageResource(imageId);
        imgIcon.setColorFilter(ContextCompat.getColor(imgIcon.getContext(), itemSelected ? R.color.defaultSurveyButton : R.color.font));
    }
}

File: examples_android/app/src/main/java/com/confirmit/testsdkapp/layouts/defaultlayout/cells/DefaultLayoutClickLabelCell.kt
package com.confirmit.testsdkapp.layouts.defaultlayout.cells

import android.view.View
import android.widget.ImageView
import android.widget.TextView
import androidx.constraintlayout.widget.ConstraintLayout
import androidx.constraintlayout.widget.ConstraintSet
import androidx.core.content.ContextCompat
import com.confirmit.testsdkapp.AppConfigs
import com.confirmit.testsdkapp.R
import com.confirmit.testsdkapp.layouts.defaultlayout.items.DefaultLayoutClickLabel
import com.confirmit.testsdkapp.layouts.defaultlayout.items.DefaultLayoutClickLabelType
import com.confirmit.testsdkapp.layouts.defaultlayout.items.DefaultLayoutItem

class DefaultLayoutClickLabelCell(view: View) : DefaultLayoutCell(view), View.OnClickListener {

    private lateinit var listener: DefaultLayoutCellChangeListener
    private lateinit var item: DefaultLayoutClickLabel
    private var itemSelected: Boolean = false

    private val btnContainer: ConstraintLayout = view.findViewById(R.id.containerButton)
    private val lblText: TextView = view.findViewById(R.id.txtLabel)
    private val imgIcon: ImageView = view.findViewById(R.id.imgIcon)

    init {
        btnContainer.setOnClickListener(this)
    }

    override fun setup(item: DefaultLayoutItem, listener: DefaultLayoutCellChangeListener) {
        this.item = item as DefaultLayoutClickLabel
        this.listener = listener

        if (item.answer.text.hasStyle && AppConfigs.htmlText) {
            lblText.text = item.answer.text.getSpanned()
        } else {
            lblText.text = item.answer.text.get()
        }

        val density = imgIcon.resources.displayMetrics.density
        val set = ConstraintSet()
        set.clone(btnContainer)
        set.setMargin(imgIcon.id, ConstraintSet.START, (density * 24 + item.indent * (density * 24)).toInt())
        set.applyTo(btnContainer)

        itemSelected = listener.getSelected(item.answer)

        setIcon()
    }

    override fun onClick(view: View?) {
        if (itemSelected && item.clickLabelType == DefaultLayoutClickLabelType.RADIO) {
            return
        }

        itemSelected = !itemSelected
        setIcon()
        when (item.clickLabelType) {
            DefaultLayoutClickLabelType.CHECKBOX -> listener.onCheckClicked(item.answer, itemSelected)
            DefaultLayoutClickLabelType.RADIO -> listener.onRadioClicked(item.answer)
        }
    }

    private fun setIcon() {
        val imageId = when (item.clickLabelType) {
            DefaultLayoutClickLabelType.CHECKBOX -> {
                if (itemSelected) R.drawable.ic_outline_check_black else R.drawable.ic_outline_box_black
            }
            DefaultLayoutClickLabelType.RADIO -> {
                if (itemSelected) R.drawable.ic_radio_black else R.drawable.ic_radio_unchecked_black
            }
        }

        imgIcon.setImageResource(imageId)
        imgIcon.setColorFilter(ContextCompat.getColor(imgIcon.context, if (itemSelected) R.color.defaultSurveyButton else R.color.font))
    }
}
